packages/suite/src/utils/suite/__tests__/parseUri.test.ts

```diff
@@ -41,4 +41,46 @@ describe('parseUri', () => {
             });
         });
     });
+
+    describe('parseUri multi', () => {
+        it('should work with any type of URIs', () => {
+            expect(parseUri('http://www.trezor.io')).toEqual({
+                address: '//www.trezor.io',
+            });
+            expect(parseUri('www.trezor.io')).toEqual({
+                address: 'www.trezor.io',
+            });
+            expect(parseUri('www.trezor.io/route')).toEqual({
+                address: 'www.trezor.io/route',
+            });
+            expect(parseUri('www.trezor.io/route?query=1&odd')).toEqual({
+                address: 'www.trezor.io/route',
+                query: '1',
+            });
+            expect(parseUri('www.trezor.io?query=1&amount=1')).toEqual({
+                address: 'www.trezor.io',
+                query: '1',
+                amount: '1',
+            });
+        });
+    });
+
+    describe('getProtocolInfo', () => {
+        it('should work with bitcoin URI', () => {
+            const protocolInfo = getProtocolInfo(
+                'bitcoin:3QmuBaZrJNCxc5Xs7aGzZUK8RirUT8jRKf?amount=0.1',
+            );
+            expect(protocolInfo).toEqual({
+                scheme: 'bitcoin',
+                amount: 0.1,
+                address: '3QmuBaZrJNCxc5Xs7aGzZUK8RirUT8jRKf',
+            });
```

Comment:
what about to add some cases with valid (not supported yet) params like `label` and `message`

`bitcoin:mjSk1Ny9spzU2fouzYgLqGUD8U41iR35QN?amount=0.10&label=Example+Merchant&message=Order+of+flowers+%26+chocolates`

and maybe even custom `foo=bar`

I believe they should be ignored and they are, just nice to have it covered by tests.

Reply:
9c2e9b22e0309afac357bd4773e0ca574efc61d5